adds reagents splashing animation

kawoppi opened the pull request:
## About The Pull Request
Adds animations for splashing reagents on stuff. Both when combat mode + right clicking something, and when throwing the reagent container at something. The slosh sound now also plays in both these cases, because it seemed weird that these were lacking sounds. There's also an animation for splashing reagents into hydroponics trays.

default splash animation: ![default](https://user-images.githubusercontent.com/94711066/209586106-1c64bef2-ce1c-40e0-af12-8b17e40efd4c.gif)
splash animation when targeting turfs: ![floor](https://user-images.githubusercontent.com/94711066/209586125-c9eee1b0-96fc-4dd2-9b7b-4715f87d8842.gif)
animation for pouring reagents into hydroponics trays: ![hydroponics](https://user-images.githubusercontent.com/94711066/209586127-669be1d0-5fb8-4ae3-a611-7bde9876dcdc.gif)
(they all get recolored by the reagents)

[new demo video!](https://youtu.be/PuoOmbc3-Nc)

~~[old demo video](https://youtu.be/SfHrINpIidk)
[old extra demo because i forgot to record it the first time](https://youtu.be/CscgltRpE64)~~
## Why It's Good For The Game
Visual and auditory feedback for splashing things and getting splashed (with potentially dangerous chemicals). Also looks neat when watering your plants.
## Changelog
:cl:
add: added reagent splashing animation for spilling reagents on things
add: added reagent splashing animation for spilling reagents onto floors
add: added reagent splashing animation for pouring reagents into hydroponics trays
soundadd: slosh sound now plays when spilling reagents on things, both directly and by throwing
/:cl:

kawoppi committed:
splashing makes an overlay appear

kawoppi committed:
added splash sound effect

kawoppi committed:
recolors splash based on reagents

kawoppi committed:
made splash sprite

kawoppi committed:
added watering animation to hydroponics trays

kawoppi committed:
animation loops reliably now!!!

thanks watermelon!

kawoppi committed:
corrected animation duration

kawoppi committed:
added animation and sound to throwing as well

Jacquerel commented:
The offset makes this look great on hydroponics trays but less great on random other turfs, it's a really good idea though.

kawoppi commented:
> The offset makes this look great on hydroponics trays but less great on random other turfs, it's a really good idea though.

There's a seperate offset for hydroponics trays. I lined up the normal animation with the middle of a floor tile since I thought it would make the most sense in terms of perspective. It's a bit late so I've lost my grasp of perspective, I'll give it another look later and see if I can improve it. I could probably have different sprites for splashing on mobs/floors so they make more sense with their perspectives.

kawoppi committed:
Added separate animations for hydroponics and turf splashes

Hydroponics is the old default. Really happy with how the floor splash turned out. The new default not so much however...

kawoppi committed:
offset hydroponics splash in sprite instead of code

might as well if i'm not using it for other splishy sploshies anymore

kawoppi committed:
improved hydroponics splash animation

ChungusGamer666 commented:
animation is a bit too slow imo

ChungusGamer666 commented on code/modules/reagents/reagent_containers.dm:
```diff
@@ -116,6 +116,14 @@
+	if(isturf(target))
+		splash_animation = image('icons/effects/effects.dmi', target, "splash_floor")
+	splash_animation.color = mix_color_from_reagents(reagents.reagent_list)
+	flick_overlay_global(splash_animation, GLOB.clients, 0.8 SECONDS)
```
should probably flick_overlay_view() but i'm not entirely sure

kawoppi replied:
from what I was told flick_overlay_global would do the same in this case because it only displays to those who can view the target the image is placed on, while also being slightly more efficient

kawoppi committed:
improved floor splash animation

kawoppi committed:
improved default splash animation

i threw a cup of water on the bathroom wall for this
